### ISSUE TITLE
Messages don't get saved to the internal SMS/MMS database

### ISSUE THREAD

**Issue by klinker24** [klinker-apps/messenger-issues#78]
**Device with issue:** Samsung Galaxy S7 running Android 6.0
**App version:** 1.2
**Steps to reproduce:**

1. Receive a message
2. View it within Messenger
3. It is NOT saved to the internal content provider

**Additional information:**

Safe to assume this probably is the case with all newer samsung devices. We must be missing a field right now.

We did change it so that it would save things to our database, however, this probably isn't sufficient. In the `SmsReceivedReceiver`, I simply catch the exception on `line 83` so that the message still gets saved to our database.

**Possible solution:**

It is working in evolve, so it is possible to fix this. Just not sure how yet.

**Comment by klinker41** [comment]
My best guess to this is that one of the required columns on Samsung devices isn't getting filled when inserting it. That could cause a null pointer when trying to cast the null value to int. We can look at evolve's code for this and see what we're missing in the sms receiver.

**Comment by klinker41** [comment]
Changed the insert method slightly with https://github.com/klinker41/messenger/commit/379c927e72acc159e5ddfc4f6cd0295561d34726.

In Evolve, we insert the values with the following:

```
 ContentValues values = new ContentValues();
values.put("address", address);
values.put("body", body);
values.put("date", dateReceived);
values.put("read", blacklistType == 1 ? "1" : "0");
values.put("date_sent", date);
Uri uri = this.getContentResolver().insert(Uri.parse("content://sms/inbox"), values);
```

which was inserting the read as a string instead of boolean. SQLite doesn't support booleans anyways, so maybe this is what was giving us the issue? It could just be trying to parse a string into an int and that won't work when the string is "true", but it will when it's "0". In messenger I've changed it to `values.put(Telephony.Sms.READ, "0");`, so we'll have to monitor to see if that helps at all.

@klinker24 test on Hannah's phone when you get the chance, this should now be identical to what Evolve does.
